src/test/java/com/google/step/youtube/DescriptionParserTest.java

```diff
@@ -334,5 +345,165 @@ public void codeWithQuotes_multipleMatches() {
+    }
+
+    @Test
+    public void toAtLinks_multipleMatchesHttpsAndHttp() {
+        String desc = "Go to https://NordVPN.com/pewdiepie and use code PEWDIEPIE to get 70% "
+            + "off a 3 year plan plus 1 additional month free."
+            + "Go to http://stamps.com, click on the microphone at the top of the homepage";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Arrays.asList("https://NordVPN.com/pewdiepie", "http://stamps.com")));
+    }
+
+    @Test
+    public void toAtLinks_noHttpOnlyWWW() {
+        String desc = "Go to www.NordVPN.com/pewdiepie and use code PEWDIEPIE to get 70% "
+            + "off a 3 year plan plus 1 additional month free.";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+    
+    @Test
+    public void toAtLinks_lookbehind() {
+        String desc = "Use code LINUS and get 25% off GlassWire https://lmg.gg/glasswire at";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    @Test
+    public void toAtLinks_standaloneAt() {
+        String desc = "Use code LINUS and get 25% off GlassWire what https://lmg.gg/glasswire";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    @Test
+    public void toAtLinks_standaloneTo() {
+        String desc = "Use code LINUS and get 25% off GlassWire burrito https://lmg.gg/glasswire";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    @Test
+    public void toAtLinks_keywordAtStart() {
+        String desc = "At https://lmg.gg/glasswire use code LINUS and get 25% off GlassWire";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Arrays.asList("https://lmg.gg/glasswire")));
+    }
+
+    @Test
+    public void toAtLinks_keywordAtEnd() {
+        String desc = "Use code LINUS and get 25% off GlassWire at https://lmg.gg/glasswire";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Arrays.asList("https://lmg.gg/glasswire")));
+    }
+
+    @Test
+    public void toAtLinks_2SpacesBetween() {
+        String desc = "Use code LINUS and get 25% off GlassWire at  https://lmg.gg/glasswire";
+
+        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(actual, equalTo(Collections.emptyList()));
+    }
+
+    @Test
+    public void toAtLinks_newLineBetween() {
+        String desc = "Use code LINUS and get 25% off GlassWire at\nhttps://lmg.gg/glasswire";
+
```

Comment:
optional: I would remove these blank lines between the description and the actual variable.  I think it hurts readability.

Reply:
+1

Reply:
Removed.